### PR TITLE
MAINT: add OTelTraceRawProcessor from trace ingestion migration branch

### PR DESCRIPTION
**Pull request by chenqi0805**
Signed-off-by: Qi Chen <19492223+chenqi0805@users.noreply.github.com>

### Description
This PR adds in otel-trace-raw-processor which is otel-trace-raw-prepper in https://github.com/opensearch-project/data-prepper/tree/maint/546-migrate-trace-analytics-to-event-model. 

Beside renaming plugins, some cleanups are essential:

* README: prepper -> processor
* package name in new modules: prepper -> processor
* implement interface: prepper -> processor
 
### Issues Resolved
Contributes to #1158 
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/otel-trace-raw-processor/src/main/java/com/amazon/dataprepper/plugins/processor/oteltrace/OTelTraceRawProcessor.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        final int numProcessWorkers = pluginSetting.getNumberOfProcessWorkers();
+        traceIdTraceGroupCache = CacheBuilder.newBuilder()
+                .concurrencyLevel(numProcessWorkers)
+                .maximumSize(OtelTraceRawProcessorConfig.MAX_TRACE_ID_CACHE_SIZE)
+                .expireAfterWrite(OtelTraceRawProcessorConfig.DEFAULT_TRACE_ID_TTL_SEC, TimeUnit.SECONDS)
+                .build();
+    }
+
+    /**
+     * execute the prepper logic which could potentially modify the incoming record. The level to which the record has
```
Comment (cmanning09) [inline data-prepper-plugins/otel-trace-raw-processor/src/main/java/com/amazon/dataprepper/plugins/processor/oteltrace/OTelTraceRawProcessor.java:66]: Please update the javadocs. The current form is a duplicate of the interface and references "prepper"

File: data-prepper-plugins/otel-trace-raw-processor/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+processor:
+    - otel_trace_raw:
+```
+
+## Configuration
+
+* `trace_flush_interval`: An `int` represents the time interval in seconds to flush all the descendant spans without any root span. Default to 180.
+
+## Metrics
+Apart from common metrics in [AbstractPrepper](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-api/src/main/java/com/amazon/dataprepper/model/prepper/AbstractPrepper.java), otel-trace-raw-processor introduces the following custom metrics.
```
Comment (sshivanii) [inline data-prepper-plugins/otel-trace-raw-processor/README.md:17]: I believe this should be `AbstractProcessor`

**Review by cmanning09 — DISMISSED** [review]

File: data-prepper-plugins/otel-trace-raw-processor/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+processor:
+    - otel_trace_raw:
+```
+
+## Configuration
+
+* `trace_flush_interval`: An `int` represents the time interval in seconds to flush all the descendant spans without any root span. Default to 180.
+
+## Metrics
+Apart from common metrics in [AbstractProcessor](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractProcessor.java), otel-trace-raw-processor introduces the following custom metrics.
```
[the remark this answers is not in the document]
Comment (chenqi0805) [reply data-prepper-plugins/otel-trace-raw-processor/README.md:17]: Great catch! So the new processor has offloaded all previous custom metrics onto otel-trace-source. I will rephrase the documentation here.

**Review by cmanning09 — APPROVED** [review]

**Review by sshivanii — APPROVED** [review]
Thanks for making the changes

**Pull request merged**